game.js

```diff
@@ -0,0 +1,148 @@
+let bot = require("./bot.js");
+let helpers = require("./helpers.js");
+let lastLetter = "";
+let cities = [
+  "Архангельск",
+  "Борисполь",
+  "Воронеж",
+  "Гюмри",
+  "Донецк",
+  "Енакиево",
+  "Жуковский",
+  "Запорожье",
+  "Изюм",
+  "Йошкар-Ола",
+  "Киев",
+  "Лондон",
+  "Москва",
+  "Николаев",
+  "Орёл",
+  "Париж",
+  "Ростов",
+  "Селидово",
+  "Трускавец",
+  "Уфа",
+  "Флоренция",
+  "Харьков",
+  "Цюрих",
+  "Челябинск",
+  "Шанхай",
+  "Щёлково",
+  "Ыгдыр",
+  "Электросталь",
+  "Юрмала",
+  "Ялта"
+].map(item => item.toLowerCase());
+
+exports.session = class Session {
+  constructor(id, chatId = 0) {
+    this.id = id;
+    this.chatId = chatId;
+    this.spentCities = new Set();
+  }
+};
+
+function randomCity(arrCities) {
+  return arrCities[helpers.getRandomInt(arrCities.length)];
+}
+
+function lastValidLetter(str) {
+  let lastLetter = str[str.length - 1];
+  let invalidLetters = ["ь", "Ъ"];
+  let i = 2;
+
+  while (invalidLetters.includes(lastLetter) || str.length == i) {
```

Comment:
This will not work as expected, because your `lastLetter` is not changing. It is static variable

Reply:
```suggestion
    let lastLetter = str[str.length - 1];
  let invalidLetters = ["ь", "ъ"];
  let i = 2;

  while (invalidLetters.includes(lastLetter)) {
    lastLetter = str[str.length - i];
    i++;
```
I find an error in invalidLetters, one of symbol was in upper case.
Then i removed second condition from cycle